src/plugins/kbd.c

```diff
@@ -1415,13 +1450,12 @@ BDKBDBcacheStats* bd_kbd_bcache_status (const gchar *bcache_device, GError **err
         /* no cache, nothing more to get */
         return ret;
 
-    path = g_strdup_printf ("/sys/block/%s/bcache/cache/block_size", bcache_device);
-    ret->block_size = get_number_from_file (path, error);
-    g_free (path);
-    if (*error) {
-        g_clear_error (error);
+    ret->block_size = get_bcache_block_size (bcache_device, &loc_error);
+    if (loc_error) {
         g_set_error (error, BD_KBD_ERROR, BD_KBD_ERROR_BCACHE_INVAL,
-                     "Failed to get 'block_size' for '%s' Bcache device", bcache_device);
+                     "Failed to get 'block_size' for '%s' Bcache device: %s",
+                     bcache_device, loc_error->message);
```

Comment:
This can use the ``g_prefix_error()`` call, AFAICT.

Reply:
Changed to `g_propagate_prefixed_error`